### PR TITLE
fixes shad0w #3035

### PR DESCRIPTION
**Pull request by dualfade**
please review --

### CODE REVIEW

**Comment by Edu4rdSHL** [comment]
Hey @dualfade, thank you a lot, the style and PKGBUILD are perfect, anyway I'm having a problem when testing in a clean chroot env:

```
─▶ ba-dev -p shad0w-368.f4f7162-1-any.pkg.tar.zst -e "shad0w listen"
loading packages...
warning: shad0w-368.f4f7162-1 is up to date -- reinstalling
resolving dependencies...
looking for conflicting packages...

Packages (1) shad0w-368.f4f7162-1

Total Installed Size:  31.11 MiB
Net Upgrade Size:       0.00 MiB

:: Proceed with installation? [Y/n] 
(1/1) checking keys in keyring                     [######################] 100%
(1/1) checking package integrity                   [######################] 100%
(1/1) loading package files                        [######################] 100%
(1/1) checking for file conflicts                  [######################] 100%
(1/1) checking available disk space                [######################] 100%
:: Processing package changes...
(1/1) reinstalling shad0w                          [######################] 100%
:: Running post-transaction hooks...
(1/1) Arming ConditionNeedsUpdate...
Package shad0w-368.f4f7162-1-any.pkg.tar.zst installed correctly! Testing it now...
cp: cannot stat '/root/shad0w/beacon/lib/*': No such file or directory
Traceback (most recent call last):
  File "/usr/share/shad0w/shad0w.py", line 197, in <module>
    asyncio.run(shad0w.start())
  File "/usr/share/shad0w/shad0w.py", line 80, in start
    asyncio.run(tools.compile_and_store_static(self))
  File "/usr/lib/python3.9/asyncio/runners.py", line 44, in run
    return loop.run_until_complete(main)
  File "/usr/lib/python3.9/asyncio/base_events.py", line 642, in run_until_complete
    return future.result()
  File "/usr/share/shad0w/lib/tools.py", line 72, in compile_and_store_static
    buildtools.clone_source_files(rootdir="injectable", builddir=build_dir_name)
  File "/usr/share/shad0w/lib/buildtools.py", line 27, in clone_source_files
    os.chdir(basedir)
FileNotFoundError: [Errno 2] No such file or directory: '/root/shad0w/beacon'
```
can you check this?

**Comment by dualfade** [comment]
Let me double check.Listen requires sudo / root as it listens on port 443 by default. Ill ping back. Thanks !

**Comment by dualfade** [comment]
@Edu4rdSHL This is hardcode -
https://github.com/bats3c/shad0w/blob/master/shad0w.py#L84

Are we good to make a patch ? -> /usr/share/shad0w/
Let me know. 
Thanks.

**Comment by Edu4rdSHL** [comment]
As I suspected it was harcoded, yeah, make a patch and test it, if this works then submit it! Thank you

**Comment by dualfade** [comment]
Awesome. Will do. Thanks !

**Comment by Edu4rdSHL** [comment]
Looking the code, we can even remove that line. Try removing this and see if works as we already change the dir in the script that's put in /usr/bin

**Comment by dualfade** [comment]
@Edu4rdSHL If you want to hold that is fine. Will containerize when needed as I have already done.

Thank you.,

**Comment by dualfade** [comment]
@Edu4rdSHL 
Lets close this pull request ? Agree ?

**Comment by Edu4rdSHL** [comment]
Let's convert it to draft :)

**Comment by dualfade** [comment]
Ok awesome ! I will get  back to it in a bit. Rad. I'm game ;)

File: packages/shad0w/PKGBUILD [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+
+  find . -type f -name '*.pyc' -exec rm -r {} \;
+  for i in `grep -r -i '/root/shad0w' * | cut -d ":" -f 1 | uniq`; \
+    do sed -i 's/\/root\/shad0w/\/usr\/share\/shad0w/g' $i; done
+}
+
+package() {
+  cd $pkgname
+  
+  wget https://packages.microsoft.com/config/ubuntu/20.10/packages-microsoft-prod.deb \
```
Comment (Edu4rdSHL) [inline packages/shad0w/PKGBUILD:36]: Please put this in `sources()`

File: packages/shad0w/PKGBUILD [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+    -O /tmp/packages-microsoft-prod.deb
+  ar x /tmp/packages-microsoft-prod.deb --output=/tmp/
```
Comment (Edu4rdSHL) [inline packages/shad0w/PKGBUILD:38]: It needs to be in `prepare()`

File: packages/shad0w/PKGBUILD [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+
+  echo $(git rev-list --count HEAD).$(git rev-parse --short HEAD)
+}
+
+prepare() {
+  cd $pkgname
+
+  find . -type f -name '*.pyc' -exec rm -r {} \;
+  for i in `grep -r -i '/root/shad0w' * | cut -d ":" -f 1 | uniq`; \
+    do sed -i 's/\/root\/shad0w/\/usr\/share\/shad0w/g' $i; done
```
Comment (Edu4rdSHL) [inline packages/shad0w/PKGBUILD:30]:
```
  find ./ -iname *.pyc | xargs rm
  grep -Irl '/root/shad0w/' ./ | xargs sed -i 's/\/root\/shad0w/\/usr\/share\/shad0w/g'
```

**Review by Edu4rdSHL — CHANGES_REQUESTED** [review]
Please see

File: packages/shad0w/PKGBUILD (hunk 2 above)
Comment (Edu4rdSHL) [reply packages/shad0w/PKGBUILD:38]: Only line 38

**Review by Edu4rdSHL — APPROVED** [review]

**Comment by Edu4rdSHL** [comment]
@dualfade thank you a lot for it, works perfectly!

**Comment by dualfade** [comment]
> @dualfade thank you a lot for it, works perfectly!

awesome ! yeah very excited about this. Great tool. Thanks @Edu4rdSHL

**Pull request merged**